arm-analysisservices/2016-05-16/swagger/analysisservices.json

```diff
@@ -410,6 +410,11 @@
       "description": "Properties of Analysis Services resource.",
       "type": "object",
       "properties": {
+        "state": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The current state of Analysis Services resource. It allows more custom terminal state like Paused."
```

Comment:
@shihying - I would suggest providing enum values `"enum": ["array of valid values"]` and add the extension `"x-ms-enum": { "name": "State", "modelAsString": true }`. When you set it as modelAsString: true, Autorest will not treat it as an enum. Instead it will be a static class with constant properties. This still provides intellisense to the customer, but there is no validation.  Please do this for both "provisioningState" and "state".